### PR TITLE
Sign in / Sign Up

### LINKED ISSUE #32: Sign in/ Sign up component
Build a sign in/ sign up page with firebase auth.
* Users can sign up the website
* After a user is created they can sign in
* forget password
* e-mail verfication
* user can change password if they are logged in

### PR DESCRIPTION
**Pull request by gizemmdeniz**

### CODE REVIEW

**Comment by bertacsevercan** [comment]
Can you put some pictures of the components?

**Comment by gizemmdeniz** [comment]
image
image
image

File: src/App.js [hunk 1]
```diff
@@ -19,11 +21,13 @@ function App() {
           <NavBar />
           <Content style={{ background: "white", padding: "0 5px" }}>
             <div className="site-layout-content">
-              <Route path="/" component={HomePage} />
+              <Route path="/home" component={HomePage} />
```
Comment (cuygur) [inline src/App.js]: Home page is broken. I see it is empty. Do you see it ?

File: src/components/Forgot/index.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      .auth()
+      .sendPasswordResetEmail(email)
+      .then(function (user) {
+        alert("Please check your email...");
+      })
+      .catch((err) => {
+        setErrorMessage(err.message);
+        console.error(err);
+      });
+  };
```
Comment (cuygur) [inline src/components/Forgot/index.jsx]: add empty line

File: src/components/Forgot/index.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            type="submit"
+            onClick={forgotPassword}
+          >
+            {t("forgot.title.title0")}
+          </Button>
+        )}
+      </Form.Item>
+    </Form>
+  );
+}
```
Comment (cuygur) [inline src/components/Forgot/index.jsx:63]: add empty line

File: src/components/JoinUs/index.test.jsx [hunk 4]
```diff
@@ -7,7 +7,6 @@ jest.mock("react-i18next", () => ({
 }));
 
 const isTesting = true;
-
```
Comment (cuygur) [inline src/components/JoinUs/index.test.jsx]: Why did you delete this line ?

File: src/containers/Authentication/index.jsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+              Register
+            </Button>
+          </Row>
+        </Form.Item>
+        {show ? <Login /> : null}
+        {click ? <Register /> : null}
+      </div>
+    </div>
+  );
+};
```
Comment (cuygur) [inline src/containers/Authentication/index.jsx:49]: add empty line

File: src/components/Register/index.jsx [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+            type="submit"
+            onClick={addDetails}
+          >
+            {t("register.title.title0")}
+          </Button>
+        )}
+      </Form.Item>
+    </Form>
+  );
+};
```
Comment (cuygur) [inline src/components/Register/index.jsx:133]: add empty line

File: src/components/Auth/Auth.jsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import React, { useEffect, useState } from "react";
+import firebase from "../../firebaseConfig";
+
+export const AuthContext = React.createContext();
+export const AuthProvider = ({ users }) => {
+  const [currentUser, setCurrentUser] = useState(null);
+  console.log(users);
```
Comment (cuygur) [inline src/components/Auth/Auth.jsx]: delete console log

File: src/components/Login/index.jsx [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const login = (e) => {
+    e.preventDefault();
+    firebase
+      .auth()
+      .signInWithEmailAndPassword(email, password)
+      .then(() => {
+        setErrorMessage(true);
+        console.log("hi");
```
Comment (cuygur) [inline src/components/Login/index.jsx]: delete console log

File: src/containers/Authentication/index.jsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    setClick(true);
+    setShow(false);
+  };
+
+  return (
+    <div>
+      <div>
+        <Form.Item style={{ marginBottom: "5em" }} {...layout} {...tailLayout}>
+          <Row>
+            <Button
```
Comment (cuygur) [inline src/containers/Authentication/index.jsx]: Can you make selected button's background gray like in Figma?

**Review by cuygur — CHANGES_REQUESTED** [review]

File: public/index.html [hunk 10]
```diff
@@ -40,4 +41,6 @@
       To create a production bundle, use `npm run build` or `yarn build`.
     -->
   </body>
+  <script src="https://www.gstatic.com/firebasejs/8.0.1/firebase-app.js"></script>
```
Comment (bertacsevercan) [inline public/index.html]: Are you sure these are necessary? Because we should be getting everything from firebaseconfig? Can @halitbatur confirm?

File: src/components/NavBar/index.jsx [hunk 11]
```diff
@@ -52,6 +52,9 @@ const NavBar = ({ isTesting }) => {
               <Icon style={{ transform: "scale(2)" }} component={AdminSvg} />
             </Link>
           </Menu.Item>
+          <Menu.Item key="6" className="navbarLogin">
+            <Link to="/register">Register</Link>
```
Comment (bertacsevercan) [inline src/components/NavBar/index.jsx]: Can you change this to sign-in/sign-up, or maybe use a conditional to show either of them? And make sure to add a sign out button somewhere too after the user logs in.

**Review by bertacsevercan — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (gizemmdeniz) [reply src/App.js]: I added path as /homepage, otherwise route is not working. @halitbatur do you have another advise for this issue? if I keep homepage as empty the other route is not working.

File: src/components/JoinUs/index.test.jsx (hunk 4 above)
Comment (gizemmdeniz) [reply src/components/JoinUs/index.test.jsx]: i didnt delete
image

File: src/components/NavBar/index.jsx (hunk 11 above)
Comment (gizemmdeniz) [reply src/components/NavBar/index.jsx]: I changed the name as Sign-In / Sign-Up for now. Maybe we can change the icon and we can move the sign in sign out section on the right side of the navbar.

File: src/components/JoinUs/index.test.jsx (hunk 4 above)
Comment (cuygur) [reply src/components/JoinUs/index.test.jsx]: there was empty line

File: src/App.js (hunk 1 above)
Comment (bertacsevercan) [reply src/App.js]: The "/" is the default path for a website so it's the first page that we see. In our case, the homepage is the default page so the path should be "/" for it IMO. Wondering what @halitbatur has to say?

**Comment by bertacsevercan** [comment]
The CI tests are not passing, you forgot to run yarn fix maybe?

File: public/index.html (hunk 10 above)
Comment (halitbatur) [reply public/index.html]: these are not necessary

File: src/components/Forgot/index.jsx [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+  const [email, setEmail] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const forgotPassword = (e) => {
+    setErrorMessage(null);
+    firebase
+      .auth()
+      .sendPasswordResetEmail(email)
+      .then(function (user) {
+        alert("Please check your email...");
```
Comment (halitbatur) [inline src/components/Forgot/index.jsx]: lets not use the alert function here it doesn't look professional maybe there's a component in antd for alerts try to use this one

File: src/components/Forgot/index.jsx [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+      {errorMessage ? (
+        <Alert
+          message="You Can start add your coupon"
+          description="Success"
+          type="success"
+        />
+      ) : null}
+      {!errorMessage ? <p>{errorMessage}</p> : null}
+      <Form.Item
+        rules={[{ required: true, message: "Please input your username!" }]}
```
Comment (halitbatur) [inline src/components/Forgot/index.jsx]: the message says please input ur username but you need the email here

File: src/components/Forgot/index.jsx [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+      >
+        <Input
+          prefix={<MailOutlined />}
+          style={{ width: "40em", display: "flex" }}
+          placeholder={t("forgot.title.title1")}
+          action="click"
+          type="email"
+          className="ant-input"
+          value={email}
+          onChange={(e) => setEmail(e.target.value)}
```
Comment (halitbatur) [inline src/components/Forgot/index.jsx:45]: im confused here why there's 2 inputs that are getting the user email ?

File: src/components/Forgot/index.jsx [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+    wrapperCol: { span: 8 },
+  };
+
+  const tailLayout = {
+    wrapperCol: { offset: 8, span: 16 },
+  };
+
+  return (
+    <Form {...layout} {...tailLayout}>
+      {errorMessage ? (
```
Comment (halitbatur) [inline src/components/Forgot/index.jsx]: having this one is suffecient to render the error message line 43 is not needed

File: src/components/Login/index.jsx [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+const layout = {
+  labelCol: { span: 4 },
+  wrapperCol: { span: 8 },
+};
+
+const tailLayout = {
+  wrapperCol: { offset: 8, span: 16 },
+};
+
+const SignUp = ({ isTesting }) => {
```
Comment (halitbatur) [inline src/components/Login/index.jsx]: the directory name is Login why there's a signup component in here ?

File: src/components/Login/index.jsx [hunk 17]
```diff
@@ -0,0 +1,112 @@
...
+  const { t } = useTranslation();
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const login = (e) => {
```
Comment (halitbatur) [inline src/components/Login/index.jsx]: and there's a login function in side the signup component

File: src/components/Login/index.jsx [hunk 18]
```diff
@@ -0,0 +1,112 @@
...
+    history.push(path);
+  };
+
+  const resetPassword = (e) => {
+    routeChange();
+  };
+
+  return (
+    <Form {...layout} {...tailLayout}>
+      {errorMessage ? (
```
Comment (halitbatur) [inline src/components/Login/index.jsx]: why is it called an errorMessage if the type is success please change variable name

File: src/components/Login/index.jsx [hunk 19]
```diff
@@ -0,0 +1,112 @@
...
+          description="Success"
+          type="success"
+          showIcon
+        />
+      ) : null}
+      {!errorMessage ? <p>{errorMessage}</p> : null}
+      <Form.Item
+        rules={[{ required: true, message: "Please input your username!" }]}
+        value={email}
+        onChange={(e) => setEmail(e.target.value)}
```
Comment (halitbatur) [inline src/components/Login/index.jsx]: same thing with forget password 2 onChanges functions I think one of them is extra

File: src/components/Login/index.jsx [hunk 20]
```diff
@@ -0,0 +1,112 @@
...
+        <Alert
+          message="You Can start add your coupon"
+          description="Success"
+          type="success"
+          showIcon
+        />
+      ) : null}
+      {!errorMessage ? <p>{errorMessage}</p> : null}
+      <Form.Item
+        rules={[{ required: true, message: "Please input your username!" }]}
```
Comment (halitbatur) [inline src/components/Login/index.jsx]: asks for username while its actually the email

File: src/components/Login/index.jsx [hunk 21]
```diff
@@ -0,0 +1,112 @@
...
+          type="email"
+          className="ant-input"
+          value={email}
+          onChange={(e) => setEmail(e.target.value)}
+        />
+      </Form.Item>
+      <Form.Item
+        rules={[{ required: true, message: "Please input your password!" }]}
+        value={password}
+        onChange={(e) => setPassword(e.target.value)}
```
Comment (halitbatur) [inline src/components/Login/index.jsx]: same thing 2 onChange functions one of them is extra

File: src/components/NavBar/index.jsx [hunk 22]
```diff
@@ -52,6 +52,9 @@ const NavBar = ({ isTesting }) => {
               <Icon style={{ transform: "scale(2)" }} component={AdminSvg} />
             </Link>
           </Menu.Item>
+          <Menu.Item key="6" className="navbarLogin">
+            <Link to="/register">Sign-In/Sign-Up</Link>
```
Comment (halitbatur) [inline src/components/NavBar/index.jsx:84]: why is there 1 link that points to signin and signup page they should be seprate

File: src/components/Register/index.jsx [hunk 23]
```diff
@@ -0,0 +1,146 @@
...
+  wrapperCol: { offset: 8, span: 16 },
+};
+
+const Register = ({ isTesting }) => {
+  const { t } = useTranslation();
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [registerInfo, setRegisterInfo] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+  const [name, setOrganizationName] = useState("");
```
Comment (halitbatur) [inline src/components/Register/index.jsx]: state setState, please fix this

File: src/firebaseConfig.js [hunk 24]
```diff
@@ -1,5 +1,6 @@
-import * as firebase from "firebase";
-import "firebase/database";
+import firebase from "firebase/app";
+import "@firebase/auth";
```
Comment (halitbatur) [inline src/firebaseConfig.js]: i dont think line 2 and 3 are needed

File: src/firebaseConfig.js [hunk 25]
```diff
@@ -1,5 +1,6 @@
-import * as firebase from "firebase";
```
Comment (halitbatur) [inline src/firebaseConfig.js]: u shouldn't delete line 1 and 2

File: src/locales/en/translation.json [hunk 26]
```diff
@@ -66,5 +66,29 @@
       "title5": "The person in need can eat without asking in available restaurants",
       "title6": "The restaurant delete the used coupon in the website"
     }
+  },
+  "login": {
+    "title": {
+      "title0": "Login",
```
Comment (halitbatur) [inline src/locales/en/translation.json]: better namings here would be better

**Review by halitbatur — CHANGES_REQUESTED** [review]
few changes need to be done but good job over all,

File: src/components/Forgot/index.jsx (hunk 12 above)
Comment (bertacsevercan) [reply src/components/Forgot/index.jsx]: https://ant.design/components/message/#header
https://ant.design/components/alert/#header

**Comment by halitbatur** [comment]
@gizemcdeniz could you please attend to the comments and fix the mistakes that i mentioned

File: src/App.js (hunk 1 above)
Comment (gizemmdeniz) [reply src/App.js]: I added exact for homepage

File: public/index.html (hunk 10 above)
Comment (gizemmdeniz) [reply public/index.html]: removed

File: src/components/Forgot/index.jsx (hunk 13 above)
Comment (gizemmdeniz) [reply src/components/Forgot/index.jsx]: changed

File: src/components/Login/index.jsx (hunk 16 above)
Comment (gizemmdeniz) [reply src/components/Login/index.jsx]: changed.

File: src/firebaseConfig.js (hunk 24 above)
Comment (gizemcdeniz) [reply src/firebaseConfig.js:2]: if i delete them, the register and login are not working

Comment (gizemmdeniz) [reply src/firebaseConfig.js]: deleted

File: src/components/Auth/Auth.jsx [hunk 27]
```diff
@@ -0,0 +1,25 @@
...
+export const AuthContext = React.createContext();
+export const AuthProvider = ({ users }) => {
+  const [currentUser, setCurrentUser] = useState(null);
+
+  useEffect(() => {
+    firebase.auth().onAuthStateChanged((user) => {
+      setCurrentUser(user);
+    });
+  }, []);
+  console.log(currentUser);
```
Comment (halitbatur) [inline src/components/Auth/Auth.jsx]: no console.logs please

File: src/components/Forgot/index.jsx (hunk 14 above)
Comment (gizemmdeniz) [reply src/components/Forgot/index.jsx:45]: I forgot to delete this line now fixed.

File: src/components/Forgot/index.jsx (hunk 15 above)
Comment (gizemmdeniz) [reply src/components/Forgot/index.jsx]: deleted.

File: src/components/NavBar/index.jsx (hunk 22 above)
Comment (gizemmdeniz) [reply src/components/NavBar/index.jsx:84]: Beacause according to design, the sign in and sign out in same page and it's combined with a button.

File: src/components/Forgot/index.jsx (hunk 12 above)
Comment (bertacsevercan) [reply src/components/Forgot/index.jsx]: Can you use the links I mentioned above for the alerts, use antd message maybe? @gizemcdeniz

File: src/components/Login/index.jsx (hunk 17 above)
Comment (gizemmdeniz) [reply src/components/Login/index.jsx]: ı changed the function name as signin

File: src/components/Login/index.jsx (hunk 19 above)
Comment (gizemmdeniz) [reply src/components/Login/index.jsx]: deleted.

File: src/containers/Authentication/index.jsx (hunk 9 above)
Comment (gizemmdeniz) [reply src/containers/Authentication/index.jsx]: done.

File: src/components/Login/index.jsx (hunk 18 above)
Comment (gizemmdeniz) [reply src/components/Login/index.jsx]: yes you are right, ı changed it.

File: src/components/Login/index.jsx (hunk 20 above)
Comment (gizemmdeniz) [reply src/components/Login/index.jsx]: i changed it as a email.

File: src/components/Login/index.jsx (hunk 21 above)
Comment (gizemmdeniz) [reply src/components/Login/index.jsx]: deleted.

File: src/components/Auth/Auth.jsx [hunk 28]
```diff
@@ -0,0 +1,25 @@
+import React, { useEffect, useState } from "react";
+import firebase from "../../firebaseConfig";
+
+export const AuthContext = React.createContext();
+export const AuthProvider = ({ users }) => {
```
Comment (halitbatur) [inline src/components/Auth/Auth.jsx]: you did this authProvider function but i dont see it getting used anywhere ?

File: src/components/Forgot/index.jsx [hunk 29]
```diff
@@ -0,0 +1,73 @@
...
+import firebase from "../../firebaseConfig";
+import { MailOutlined } from "@ant-design/icons";
+import { useTranslation } from "react-i18next";
+
+function Forgot({ isTesting }) {
+  const { t } = useTranslation();
+  const [email, setEmail] = useState("");
+  const [successMessage, setSuccessMessage] = useState("");
+
+  const forgotPassword = (e) => {
```
Comment (halitbatur) [inline src/components/Forgot/index.jsx]: here e is not getting used so we can just remove it

File: src/components/Forgot/index.jsx [hunk 30]
```diff
@@ -0,0 +1,73 @@
...
+    wrapperCol: { span: 8 },
+  };
+
+  const tailLayout = {
+    wrapperCol: { offset: 8, span: 16 },
+  };
+
+  return (
+    <Form {...layout} {...tailLayout}>
+      {successMessage ? (
```
Comment (halitbatur) [inline src/components/Forgot/index.jsx]: I think this one is enough for showing sucess message, could u please delete line 41

File: src/components/Forgot/index.jsx (hunk 13 above)
Comment (halitbatur) [reply src/components/Forgot/index.jsx]: @gizemcdeniz still not changed

File: src/components/Login/index.jsx [hunk 31]
```diff
@@ -0,0 +1,110 @@
...
+  wrapperCol: { offset: 8, span: 16 },
+};
+
+const Login = ({ isTesting }) => {
+  const { t } = useTranslation();
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [successMessage, setSuccessMessage] = useState("");
+
+  const signin = (e) => {
```
Comment (halitbatur) [inline src/components/Login/index.jsx:19]: e is not getting used here

File: src/components/Login/index.jsx [hunk 32]
```diff
@@ -0,0 +1,110 @@
...
+  };
+
+  const resetPassword = (e) => {
+    routeChange();
+  };
+
+  return (
+    <div className="formInputSection" {...layout} {...tailLayout}>
+      <Form>
+        {successMessage ? (
```
Comment (halitbatur) [inline src/components/Login/index.jsx]: I think for the seccess message you can do it with one if condition so please delete line 57

File: src/components/Register/index.jsx [hunk 33]
```diff
@@ -0,0 +1,157 @@
...
+  const addOrganization = (e) => {
+    db.collection("restaurants").doc(uid.current).set({
+      name: name,
+      email: email,
+      uid: uid.current,
+    });
+    setOrganizationName("");
+  };
+
+  const handleClick = (e) => {
```
Comment (halitbatur) [inline src/components/Register/index.jsx]: e is not getting used here

File: src/components/Register/index.jsx [hunk 34]
```diff
@@ -0,0 +1,157 @@
...
+        setRegisterInfo(true);
+      })
+      .catch((err) => {
+        setRegisterInfo(false);
+        setErrorMessage(err.message);
+        console.error(err);
+      });
+  };
+
+  let history = useHistory();
```
Comment (halitbatur) [inline src/components/Register/index.jsx:48]: new line after line 62

File: src/components/Register/index.jsx [hunk 35]
```diff
@@ -0,0 +1,157 @@
...
+  const addDetails = (e) => {
+    addOrganization(e);
+    handleClick(e);
+    routeChange();
+  };
+
+  return (
+    <div className="formInputSection" {...layout} {...tailLayout}>
+      <Form>
+        {registerinfo ? (
```
Comment (halitbatur) [inline src/components/Register/index.jsx]: same here, line 85 is not needed

File: src/containers/Authentication/index.jsx [hunk 36]
```diff
@@ -0,0 +1,60 @@
+import React, { useState } from "react";
+import Register from "../../components/Register";
+import Login from "../../components/Login";
+import { Form, Row, Button } from "antd";
+import "./index.css";
+
+const layout = {
```
Comment (halitbatur) [inline src/containers/Authentication/index.jsx]: I see these are used alot maybe we can move it to another file and import it from there instead of writing it everytime in the component ?

File: src/containers/Authentication/index.jsx [hunk 37]
```diff
@@ -0,0 +1,60 @@
...
+              type="inner"
+              onClick={handleClick}
+              value="vertical"
+            >
+              Register
+            </Button>
+          </Row>
+        </Form.Item>
+        {show ? <Login /> : null}
+        {click ? <Register /> : null}
```
Comment (halitbatur) [inline src/containers/Authentication/index.jsx]: why not just 1 state that renders either login or register ?

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/containers/Authentication/index.jsx (hunk 37 above)
Comment (gizemmdeniz) [reply src/containers/Authentication/index.jsx]: can you show me an example with just one state? I can not do it just one.

**Comment by halitbatur** [comment]
@gizemcdeniz can u fix conflicts please

File: src/components/Forgot/index.jsx [hunk 38]
```diff
@@ -0,0 +1,72 @@
...
+    firebase
+      .auth()
+      .sendPasswordResetEmail(email)
+      .then(function (user) {
+        setSuccessMessage("Please check your email...");
+      })
+      .catch((err) => {});
+  };
+
+  const layout = {
```
Comment (halitbatur) [inline src/components/Forgot/index.jsx]: why is this still here and not imported  from the layout file ?

File: src/components/Register/index.jsx (hunk 23 above)
Comment (halitbatur) [reply src/components/Register/index.jsx]: @gizemcdeniz still not fixed here the state name should be organizationName

File: src/components/Register/index.jsx [hunk 39]
```diff
@@ -0,0 +1,154 @@
...
+  const [registerinfo, setRegisterInfo] = useState(false);
+  const [errormessage, setErrorMessage] = useState("");
+  const [name, setOrganizationName] = useState("");
+  const uid = useRef();
+
+  firebase.auth().onAuthStateChanged(function (user) {
+    if (user) {
+      uid.current = user.uid;
+    } else {
+      console.log("err");
```
Comment (halitbatur) [inline src/components/Register/index.jsx]: lets not use console.log lets use alerts please

File: src/components/Register/index.jsx [hunk 40]
```diff
@@ -0,0 +1,154 @@
...
+    }
+  });
+
+  const addOrganization = (e) => {
```
Comment (halitbatur) [inline src/components/Register/index.jsx]: here the e is not used lets remove it

File: src/components/Register/index.jsx [hunk 41]
```diff
@@ -0,0 +1,154 @@
...
+          </Checkbox>
+        </Form.Item>
+        <Form.Item>
+          {isTesting ? (
+            <button>Donate</button>
+          ) : (
+            <Button
+              style={{ background: "#2F80ED" }}
+              type="submit"
+              onClick={addDetails}
```
Comment (halitbatur) [inline src/components/Register/index.jsx]: lets remove the onClick function and add on Submit for the whole form in that case u will need to use e.preventDefault()

File: src/components/Register/index.jsx [hunk 42]
```diff
@@ -0,0 +1,154 @@
...
+      name: name,
+      email: email,
+      uid: uid.current,
+    });
+    setOrganizationName("");
+  };
+
+  const handleClick = () => {
+    setErrorMessage(null);
+    // e.preventDefault();
```
Comment (halitbatur) [inline src/components/Register/index.jsx]: lets uncomment this and call this function for the onSubmit not the onClick

File: src/containers/Authentication/index.jsx (hunk 37 above)
Comment (halitbatur) [reply src/containers/Authentication/index.jsx]: ``` show ? <Login /> : <Register /> ```

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/components/Register/index.jsx (hunk 41 above)
Comment (gizemmdeniz) [reply src/components/Register/index.jsx]: I tried it but it's not working, i put this function on form as well.
image

File: public/index.html [hunk 43]
```diff
@@ -27,6 +27,7 @@
     <title>React App</title>
   </head>
   <body>
+
```
Comment (bertacsevercan) [inline public/index.html]: Maybe remove this empty line?

File: src/components/Forgot/index.jsx (hunk 13 above)
Comment (bertacsevercan) [reply src/components/Forgot/index.jsx]: Can you change the message here to "Please input your email!" ? And use a translation hook.

File: src/components/Register/index.jsx [hunk 44]
```diff
@@ -0,0 +1,136 @@
...
+import { Form, Input, Button, Checkbox, message } from "antd";
+import { useHistory } from "react-router-dom";
+import firebase from "../../firebaseConfig";
+import { useTranslation } from "react-i18next";
+import { MailOutlined } from "@ant-design/icons";
+import "./index.css";
+import { Layout } from "../Layout";
+
+const db = firebase.firestore();
+
```
Comment (bertacsevercan) [inline src/components/Register/index.jsx]: Please remove this empty line.

File: src/components/Register/index.jsx (hunk 23 above)
Comment (bertacsevercan) [reply src/components/Register/index.jsx]: Yes you need to change the state to organizationName. `const [organizationName, setOrganizationName] = useState("");`

Comment (bertacsevercan) [reply src/components/Register/index.jsx]: or change the setter to = setName

File: src/firebaseConfig.js (hunk 25 above)
Comment (bertacsevercan) [reply src/firebaseConfig.js]: You need to add these back: `import * as firebase from "firebase";
import "firebase/database";`

**Review by bertacsevercan — CHANGES_REQUESTED** [review]

**Review by bertacsevercan — APPROVED** [review]
LGTM! But you need to solve the conflicts and be sure to check Halit's comments too and ask a re-review if you did them!

**Review by bertacsevercan — APPROVED** [review]

File: src/App.js [hunk 45]
```diff
@@ -22,10 +24,13 @@ function App() {
           <Content style={{ background: "white", padding: "0 5px" }}>
             <div className="site-layout-content">
               <Route exact path="/" component={HomePage} />
+              <Route path="/coupons" component="#" />
```
Comment (halitbatur) [inline src/App.js]: can we delete this line ? we have already changed the component its rendering

File: src/components/Forgot/index.jsx [hunk 46]
```diff
@@ -0,0 +1,64 @@
...
+    firebase
+      .auth()
+      .sendPasswordResetEmail(email)
+      .then(() =>
+        message.success(t("forgotEmail.forgotEmailtitle.forgotEmailSuccess"))
+      )
+      .catch(() =>
+        message.warning(t("forgotEmail.forgotEmailtitle.forgotEmailError"))
+      );
+  };
```
Comment (halitbatur) [inline src/components/Forgot/index.jsx:25]: new line here

File: src/containers/Authentication/index.css [hunk 47]
```diff
@@ -0,0 +1,27 @@
+.registerButton:hover {
+  background-color: lightgrey;
+  color: black;
+}
```
Comment (halitbatur) [inline src/containers/Authentication/index.css:4]: new line here

**Review by halitbatur — APPROVED** [review]
just a few changes and its ready to be merged, you will need to fix conflitcs too

File: src/App.js [hunk 48]
```diff
@@ -27,6 +29,9 @@ function App() {
               <Route path="/blog" component="#" />
               <Route path="/cart" component={Cart} />
               <Route path="/admin" component={AdminDashboard} />
+              <Route path="/cart" component="#" />
```
Comment (halitbatur) [inline src/App.js]: why did you delete both of them.. we had a route for CouponsPage this shouldn't be deleted, and the route on line 32 should be deleted cause at line 30 we already have a route for cart

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 48 above)
Comment (halitbatur) [reply src/App.js]: I have commented on this route for my last review but you deleted other things and not the thing i commented on

**Review by halitbatur — APPROVED** [review]
